### PR TITLE
kubectl: deprecate --delete-local-data

### PR DESCRIPTION
**Pull request by dougsland**
The option --delete-local-data according with users is not clear.
This patch deprecate --delete-local-data in favor of --delete-emptydir-data.

Reference:
https://github.com/kubernetes/kubernetes/issues/80228

Signed-off-by: Douglas Schilling Landgraf <dougsland@redhat.com>

/kind feature
/king bug
/kind deprecation

**What this PR does / why we need it**:
Improve user experience.

**Which issue(s) this PR fixes**:
Fixes #https://github.com/kubernetes/kubernetes/issues/80228

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
kubectl: deprecate --delete-local-data
```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/drain/drain.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ type Helper struct {
 	GracePeriodSeconds  int
 	IgnoreAllDaemonSets bool
 	Timeout             time.Duration
-	DeleteLocalData     bool
```
Comment (eddiezane) [inline staging/src/k8s.io/kubectl/pkg/drain/drain.go:55]: 🤔 I wonder if removing this counts as a breaking change.

**Review by zhouya0 — COMMENTED** [review]
Maybe we should have a test using ` delete-local-data` to ensure backward compatity.

File: staging/src/k8s.io/kubectl/pkg/drain/drain.go (hunk 1 above)
Comment (dougsland) [reply staging/src/k8s.io/kubectl/pkg/drain/drain.go:55]: Hey @eddiezane thanks for the review. Good question, but I am wondering which programs (other than tests) would use Helper struct?

**Comment by dougsland** [comment]
> Maybe we should have a test using ` delete-local-data` to ensure backward compatity.

sounds a good plan.

**Review by soltysh — APPROVED** [review]
/approve

**Comment by dougsland** [comment]
Hey @zhouya0  @jingax10, just a friendly ping. :)

**Comment by eddiezane** [comment]
Pinging @kubernetes/sig-scalability-pr-reviews for approval

**Comment by tosi3k** [comment]
This PR doesn't look like anything scalability-related.

/assign @cjcullen

**Review by soltysh — APPROVED** [review]
/priority backlog
/approve

**Comment by eddiezane** [comment]
@tosi3k `sig-scalability-approvers` is listed in the OWNERS file for the gce script that was modified.

**Comment by mm4tt** [comment]
/approve
for /cluster/gce changes

**Comment by dougsland** [comment]
The error not related. /retest

**Comment by eddiezane** [comment]
Failure might be related to https://github.com/kubernetes/kubernetes/issues/94528. I'll dig in later.

**Comment by eddiezane** [comment]
Failure was https://github.com/kubernetes/kubernetes/issues/96168

**Pull request merged**